ui/raidboss/emulator/data/NetworkLogConverterWorker.js

```diff
@@ -0,0 +1,68 @@
+import EmulatorCommon from '../EmulatorCommon.js';
+import Encounter from './Encounter.js';
+import LogEventHandler from './LogEventHandler.js';
+import NetworkLogConverter from './NetworkLogConverter.js';
+import LogRepository from './network_log_converter/LogRepository.js';
+
+onmessage = async (msg) => {
+  const logConverter = new NetworkLogConverter();
+  const localLogHandler = new LogEventHandler();
+  const repo = new LogRepository();
+
+  // Listen for LogEventHandler to dispatch fights and persist them
+  localLogHandler.on('fight', async (day, zoneId, zoneName, lines) => {
+    const enc = new Encounter(day, zoneId, zoneName, lines);
+    if (!(enc.firstPlayerAbility > 0 && enc.firstEnemyAbility > 0))
+      return;
+    postMessage({
+      type: 'encounter',
+      encounter: enc,
+      name: enc.combatantTracker.getMainCombatantName(),
+    });
+  });
+
+  // Convert the message manually due to memory issues with extremely large files
+  const decoder = new TextDecoder('UTF-8');
+  let buf = new Uint8Array(msg.data);
+  let i = 0;
+  let lines = [];
+  let lineCount = 0;
+  for (let a = i; i < buf.length && i !== -1; a = i) {
```

Comment:
Sorry, too many single letter variables here.  What are `a` and `i`?

Reply:
`a` is the advancing index, `i` is the current index. I guess these could be renamed to `currentOffset` and `nextOffset` maybe?